### PR TITLE
mixed bag of patches - Mk install

### PR DESCRIPTION
**Pull request by jimc**
heres 4 patches

b1a33c (HEAD -> mk-install, ghlinux/mk-install) configkernel: be more like make O=outdir
b9df537 configkernel: config comment cosmetics
261e801 vng-run: get kdir from O=
2e37ad9 add Makefile install target

some issues, mostly minor, maybe some revisions ?

### CODE REVIEW

**Comment by jimc** [comment]
further testing suggests that one of these,
or the build-script using it, isnt quite ready

**Review by arighi — APPROVED** [review]
Good job @jimc ! Sorry for my late review (I've been a bit swamped in the past days).

I left a few comments, but overall it looks good. Let me know what you think. Thanks!

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,8 @@
 .PHONY: init
 init:
 	cd virtme_ng_init && cargo install --path . --root ../virtme/guest
+
+# see README.md '* Install from source'
+install: install_from_source
+install_from_source:
+	BUILD_VIRTME_NG_INIT=1 pip3 install --verbose -r requirements.txt .
```
Comment (arighi) [inline Makefile:8]: Can we add something like the following?
```
diff --git a/Makefile b/Makefile
index 23f76a3..6925d8c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+INSTALL_ARGS :=
+
 .PHONY: init
 init:
 	cd virtme_ng_init && cargo install --path . --root ../virtme/guest
@@ -5,4 +7,4 @@ init:
 # see README.md '* Install from source'
 install: install_from_source
 install_from_source:
-	BUILD_VIRTME_NG_INIT=1 pip3 install --verbose -r requirements.txt .
+	BUILD_VIRTME_NG_INIT=1 pip3 install --verbose -r requirements.txt $(INSTALL_ARGS) .
```

In this way we can pass extra install arguments (such as `--break-system-packages` that is required in Ubuntu).

Comment (jimc) [reply Makefile:8]: that makes sense.
Id seen that example and meant to ask if it belonged in the Makefile

if you want to modify my patch and commit it - great.
or I could revise it.

File: virtme_ng/run.py [hunk 2]
```diff
@@ -864,7 +864,14 @@ def _get_virtme_run(self, args):
             else:
                 self.virtme_param["kdir"] = "--kimg " + args.run
         else:
-            self.virtme_param["kdir"] = "--kdir ./"
+            if args.envs:
+                print("got env")
```
[the remark this answers is not in the document]
Comment (jimc) [reply virtme_ng/run.py:868]: its less informative over time :-}
I'll review these all, maybe verbose gate them, maybe drop entirely...

File: virtme/commands/configkernel.py [hunk 3]
```diff
@@ -360,8 +362,17 @@ def do_it():
             return 1
 
     if maketarget is not None:
+        make_args = []
+        if not os.path.exists(makef):
+            if args.verbose:
+                print(f"missing {makef}, add -f $src/Makefile")
```
[the remark this answers is not in the document]
Comment (jimc) [reply virtme/commands/configkernel.py:368]: Im sure I was heading towards adding the args,
I suspect I didnt have the python for $src (falling back to perl/shell).

I'll revise out the prints -> write
probably before I actually figure out where I left off

File: virtme/commands/configkernel.py [hunk 4]
```diff
@@ -360,8 +362,17 @@ def do_it():
...
+            if os.path.exists("Makefile"):
+                # assuming we're in linux srcdir
+                make_args = ["-f", str(os.path.abspath("Makefile"))]
+                if args.verbose:
+                    print(f"adding make_args: {make_args}")
```
[the remark this answers is not in the document]
Comment (jimc) [reply virtme/commands/configkernel.py:373]: is the assumption on 370 safe ? true ?

File: Makefile (hunk 1 above)
Comment (jimc) [reply Makefile:8]: I got it

File: virtme/commands/configkernel.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jimc) [reply virtme/commands/configkernel.py:368]: uhm - I should probably read more, catch up, before typing.. :-}

**Comment by jimc** [comment]
alright, heres a refresh of the 4,

+2 maybe's - 

alias --custom to --config - maybe only in run.py  
configkernel doesnt really need the alias I gave it

pass-thru --verbose from run to configkernel

**Comment by jimc** [comment]
hold off yet - somethings not right

**Comment by jimc** [comment]
thanks for picking up the makefile bits in #142,
The other bits need more bake-time I think.
I'll work them further...

**Comment by arighi** [comment]
@jimc I was going to start reviewing this, but noticed that you closed it. :)

Let me know if you need help with this and I can take a look if you want. Thanks!

**Comment by jimc** [comment]
On Sun, Jul 28, 2024 at 12:58 AM Andrea Righi ***@***.***>
wrote:

> @jimc <https://github.com/jimc> I was going to start reviewing this, but
> noticed that you closed it. :)
>
>
ICBW, but I think you merged most of the pieces in #142
and I was gonna retest the O=outdir related bits some more.
IIRC, the O=outdir handling works from/in configkernel,
but not quite right for vng

**Pull request closed without merging**